source/evp_fetch.c

```diff
@@ -137,17 +139,17 @@ static struct fetch_data_entry fetch_entries[] = {
 #ifndef OPENSSL_NO_POLY1305
     {FETCH_MAC, OSSL_MAC_NAME_POLY1305, NULL},
 #endif
-#ifndef OPENSSL_NO_ML_KEM
+#if defined(OPENSSL_DO_PQ) && !defined(OPENSSL_NO_ML_KEM)
```

Comment:
nit: it would be clearer to nest these algos into `OPENSSL_DO_PQ` to have it only once, but fine by me